### PR TITLE
Features/enemy

### PR DESCRIPTION
**Pull request by BhavyamKrGoyal**
With Enemy spawn and shootable

### CODE REVIEW

File: Assets/scripts/Enemy/Controller/ControllerBlueEnemy.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Enemy.Controller;
+using UnityEngine;
+
+public class ControllerBlueEnemy
+{
+    public static implicit operator ControllerBlueEnemy(ControllerEnemy v)
```
Comment (mayankgrover) [inline Assets/scripts/Enemy/Controller/ControllerBlueEnemy.cs]: what's this implicit operator? 
and v? (rename it properly even if its for 1 line of code - remember serial killer) :)

File: Assets/scripts/Enemy/Controller/ControllerRedEnemy.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ControllerRedEnemy
```
Comment (mayankgrover) [inline Assets/scripts/Enemy/Controller/ControllerRedEnemy.cs]: we don't need child classes for enemy controllers and models - custom data is controlled through different scriptable objects (SO)

File: Assets/scripts/Enemy/ScriptableEnemy.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using UnityEngine;
+using UnityEditor;
+
+[CreateAssetMenu(fileName = "ScriptableEnemy", menuName = "ScriptableObj/EnemyObject", order = 0)]
+public class ScriptableEnemy : ScriptableObject
+{
+    public float health;
+    public GameObject enemyPrefab;
```
Comment (mayankgrover) [inline Assets/scripts/Enemy/ScriptableEnemy.cs]: replace GO with EnemyView class

File: Assets/scripts/Enemy/View/ViewBlueEnemy.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ViewBlueEnemy : MonoBehaviour
```
Comment (mayankgrover) [inline Assets/scripts/Enemy/View/ViewBlueEnemy.cs]: no custom views needed for now since we don't have custom behavior yet

**Review by mayankgrover — CHANGES_REQUESTED** [review]

File: Assets/scripts/Enemy/Controller/ControllerBlueEnemy.cs (hunk 1 above)
Comment (BhavyamKrGoyal) [reply Assets/scripts/Enemy/Controller/ControllerBlueEnemy.cs]: done

File: Assets/scripts/Enemy/Controller/ControllerRedEnemy.cs (hunk 2 above)
Comment (BhavyamKrGoyal) [reply Assets/scripts/Enemy/Controller/ControllerRedEnemy.cs]: Done

File: Assets/scripts/Enemy/ScriptableEnemy.cs (hunk 3 above)
Comment (BhavyamKrGoyal) [reply Assets/scripts/Enemy/ScriptableEnemy.cs]: done

File: Assets/scripts/Enemy/View/ViewBlueEnemy.cs (hunk 4 above)
Comment (BhavyamKrGoyal) [reply Assets/scripts/Enemy/View/ViewBlueEnemy.cs]: okay

**Pull request merged**